### ISSUE TITLE
Error when try to populate a model with a Function Field

### ISSUE THREAD

**Issue by dmunozgaete** [xpepermint/rawmodeljs#18]
Hi:

when you call the populate method, in a schema model with Function field type, it's simply fails =(, because the JSON.stringify(field) call before setting the value.

Example: 

````javascript
//CheckoutButtonRenderModel.ts
import { Model } from 'rawmodel';

// defining a basic model
class CheckoutButtonRenderModel extends Model {
    public payment: Function;

    public constructor(data = <any>{}) {
        super(data);

        this.defineField('payment', {
        
        });

        this.populate(data);
    }
};
export { CheckoutButtonRenderModel };
````

````javascript
let buttonModel = new CheckoutButtonRenderModel(model);
		buttonModel
			.validate()
			.then((r) => {
			},
			(err) => {
				console.error("Validation Error ", buttonModel.collectErrors()[0].errors);

			});
````

TLDR:

the error happens when you have a field with Function type, the library try to stringify a function type and the error show as:

````javascript
/*
  * Deeply applies data to the fields.
  */

  public populate (data = {}): this {
    Object.keys(data)
      .filter((n) => !!this._fields[n])
      .forEach((name) => this[name] = serialize(data[name]));

    return this;
  }
````

and in the utils.ts 

````javascript
/*
* Returns a duplicated data object (useful to remove data bindings).
*/

export function serialize (data: any) {
  return JSON.parse(JSON.stringify(data));
}
````

**Comment by xpepermint** [comment]
You are right. This module depends on [ValidatableJS](https://github.com/xpepermint/validatablejs) for type casting but skips the `Function` field type intentionally to keep separation between the data and instance methods. I guess I need to fix the documentation for this - thanks!

Please try using [field value transformations](https://github.com/xpepermint/rawmodeljs/#field-value-transformation) instead. You could also use [your own data type](https://github.com/xpepermint/rawmodeljs/#model-class) as explained in the readme (see the [defineType](https://github.com/xpepermint/rawmodeljs/blob/master/src/models.ts#L174) method).

**Comment by xpepermint** [comment]
Thanks for pointing this out. I updated the [documentation](https://github.com/xpepermint/rawmodeljs#built-in-data-types).
